### PR TITLE
[Checkbox][CheckboxGroup]: Add new components

### LINKED ISSUE #522: [Checkbox][CheckboxGroup] component
As a mineral-ui consumer, I want a Checkbox/CheckboxGroup component so that I can build forms in my app that can accept this type of input from users.

#### AC:

1. Public Checkbox and CheckboxGroup components
1. Supports tri-state - checked, unchecked, mixed
1. Can click text as well as button to toggle Checkbox
1. Works properly inside of FormField
1. Add to component status page
1. Controlled/Uncontrolled works

#### Considerations:

Styleguide - https://ca.invisionapp.com/d/main#/console/11070131/250301208/preview

### PR DESCRIPTION
**Pull request by brentertz**
### Description

* Add new `Checkbox` and `CheckboxGroup` components
* Refactor `Radio/RadioGroup` to use `Choice/ChoiceGroup`
* Add `labelPosition` and `justify` props to `Radio`
* Add some new toggle `mineral-ui-icons` - those used in `Checkbox` and `Radio`.

NOTE: This PR does not yet handle FormField caption placement above Radios/Checkboxes and has not updated Radio colors as shown in the design spec as there are still some aspects of those design changes that need addressed.

### Motivation and context

closes #522 

### Screenshots, videos, or demo, if appropriate

https://522-checkbox--mineral-ui.netlify.com

### How to test

1. View the deploy preview
1. View the Component Status page
1. View Checkbox and CheckboxGroup pages
1. View Radio and RadioGroup pages

### Types of changes

- New feature
  * New `Checkbox/CheckboxGroup` components
  * New icons in `mineral-ui-icons`
  * New `labelPosition` and `justify` props to `Radio`
- Refactor
  * Refactor `Radio/RadioGroup` to use private `Choice/ChoiceGroup`
- BREAKING CHANGES
  * Simplified `Radio/RadioGroup` theme.  Removed keys: `RadioControl_borderColor_focus, RadioControl_boxShadow_checkedFocus, RadioControl_size_jumbo, RadioIcon_*` . Renamed all `RadioGroup` theme keys.

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change

### CODE REVIEW

File: src/website/app/demos/Radio/bestPractices.js [hunk 1]
```diff
@@ -80,5 +80,25 @@ make assumptions about your users choices.`,
...
+  {
+    type: 'dont',
+    description: `Don't use when the user should be able to select multiple
+options. Use a [Checkbox](../checkbox) instead.`,
+    example: (
+      <DemoForm>
+        <FormField
+          input={RadioGroup}
+          name="wants"
+          label="What do you want in a component library?"
```
Comment (kylegach) [inline src/website/app/demos/Radio/bestPractices.js]: Love it.

File: src/website/app/demos/Radio/index.js [hunk 2]
```diff
@@ -59,31 +43,40 @@ export default [
     whenHowToUse: `Use a Radio to allow users to select a single option from a
 set.
 
-${sharedWhenHowToUse}
+Use Radios when the number of choices is fairly small.  If the number of
```
Comment (kylegach) [inline src/website/app/demos/Radio/index.js]: In case I didn't say it on the prior PR, this is really good content.

File: src/Radio/Radio.js [hunk 3]
```diff
@@ -56,231 +61,66 @@ type Props = {
...
+        theme: choiceComponentTheme(baseTheme)
       },
-
-      '&:focus,&[data-simulate-focus]': {
-        '& + span': {
-          boxShadow: theme.RadioControl_boxShadow_focus
+      {
+        name: 'Radio',
+        theme: {
+          RadioControl_borderRadius: '100%'
```
Comment (kylegach) [inline src/Radio/Radio.js]: The diff makes this hard to read, but holy crap this is lovely.

File: src/Radio/RadioGroup.js [hunk 4]
```diff
@@ -15,131 +15,83 @@
...
   /**
-   * Checked state of the radio button. Primarily for use with controlled
-   * components. If this prop is specified, an `onChange` handler must also be
-   * specified. See also: `defaultChecked`.
+   * Value of the selected Radio; primarily for use with controlled components.
+   * If this prop is specified, an `onChange` handler must also be specified.
+   * See also: `defaultChecked`.
    */
   checked?: string,
   /** Mineral Radio components */
```
Comment (kylegach) [inline src/Radio/RadioGroup.js]: We typically link component names when used in this manner.

File: src/Checkbox/Checkbox.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        name: 'Choice',
+        theme: {}
+      },
+      baseTheme
+    )
+  };
+});
+
+/**
+ * Checkbox allows users to select one or more options from a list.
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Checkbox/Checkbox.js]: I took a stab at rewording the descriptions for Checkbox/CheckboxGroup/Radio/RadioGroup.

File: src/Checkbox/CheckboxGroup.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+import Checkbox from './Checkbox';
+
+type Props = {
+  /**
+   * Array of values of the selected Checkboxes; primarily for use with
+   * controlled components. If this prop is specified, an `onChange` handler
+   * must also be specified. See also: `defaultChecked`.
+   */
+  checked?: Array<string>,
+  /** Mineral Checkbox components */
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Checkbox/CheckboxGroup.js]: Updated.

File: src/Choice/Choice.js [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+import { createStyledComponent, pxToEm } from '../styles';
+
+type Props = {
+  /**
+   * Checked state of the input. Primarily for use with controlled
+   * components. If this prop is specified, an `onChange` handler must also be
+   * specified. See also: `defaultChecked`.
+   */
+  checked?: boolean,
+  /** @Private CSS className */
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Choice/Choice.js]: Yeah, not strictly needed.  This component was originally going to be public.

File: src/Choice/Choice.js [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+
+    return {
+      alignItems: 'center',
+      backgroundColor,
+      borderColor: theme.ChoiceControl_borderColor,
+      borderRadius: theme.ChoiceControl_borderRadius,
+      borderStyle: 'solid',
+      borderWidth: '1px',
+      color: backgroundColor,
+      content: ' ',
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Choice/Choice.js]: Updated.

File: src/Choice/Choice.js [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+      display: 'inline-flex',
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Choice/Choice.js]: Updated.

File: src/Choice/Choice.js [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+      }
+    };
+  },
+  input: ({ theme: baseTheme }) => {
+    const theme = componentTheme(baseTheme);
+
+    return {
+      ...hideVisually(),
+
+      // NOTE: These hover styles are only needed for the "states" demo.
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Choice/Choice.js]: Updated. `data-simulate-hover` was not needed in the root.

File: src/Choice/ChoiceGroup.js [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+  /** Props to be applied directly to the root element */
+  rootProps?: Object,
+  /** @Private Available sizes */
+  size?: 'small' | 'medium' | 'large' | 'jumbo',
+  /** Type of component that will be rendered */
+  type: 'checkbox' | 'radio'
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  ChoiceGroup_marginHorizontal_inline: baseTheme.space_inline_xl,
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/Choice/ChoiceGroup.js]: Updated.  The inline example appears to have been removed from Invision.

File: src/Radio/Radio.js (hunk 3 above)
Comment (brentertz) [reply src/Radio/Radio.js]: Yay!

File: src/Radio/RadioGroup.js (hunk 4 above)
Comment (brentertz) [reply src/Radio/RadioGroup.js]: Updated.

File: src/website/app/demos/Checkbox/bestPractices.js [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+            { label: 'Quality', value: 'quality' },
+            { label: 'Speed', value: 'speed' }
+          ]}
+        />
+      </DemoForm>
+    )
+  },
+  {
+    type: 'do',
+    description: `For a simple boolean questions, use a single checkbox.`,
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/bestPractices.js]: Updated

File: src/website/app/demos/Checkbox/bestPractices.js [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    example: (
+      <DemoForm>
+        <Checkbox name="terms" label="I agree to the terms of service." />
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/bestPractices.js]: Agreed.  I thought I fixed that.  Updated.

File: src/website/app/demos/Checkbox/examples/checkbox-group/formField.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+import DemoForm from '../../components/DemoForm';
+
+export default {
+  id: 'form-field',
+  title: 'FormField',
+  description: `Use a [FormField](../form-field) to provide an accessible label
+and other features as well as a more streamlined API.
+
+_Note: The \`invalid\` and \`required\` props are not automatically forwarded
+to Checkboxes because doing so might not do what the author expects (marking,
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/examples/checkbox-group/formField.js]: Updated.

File: src/website/app/demos/Checkbox/examples/checkbox-group/formField.js [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+**all** Checkboxes as invalid or required).  These props can however be set
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/examples/checkbox-group/formField.js]: Updated.

File: src/website/app/demos/Checkbox/examples/checkbox/invalid.js [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import DemoForm from '../../components/DemoForm';
+import Checkbox from '../../../../../../Checkbox';
+
+export default {
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/examples/checkbox/invalid.js]: Yeah, I am not really convinced of the value of these required and invalid examples.  I feel like they could use more explanation, but cannot quite put my finger on it.  Did not update anything, but am open to more specific suggestions.

File: src/website/app/demos/Checkbox/index.js [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+const checkboxDoc = require('!!react-docgen-loader!../../../../Checkbox/Checkbox');
+const checkboxGroupDoc = require('!!react-docgen-loader!../../../../Checkbox/CheckboxGroup');
+
+export default [
+  {
+    bestPractices,
+    componentTheme: checkboxComponentTheme,
+    doc: checkboxDoc,
+    examples: checkboxExamples,
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/index.js]: Took a stab at updating.  Moved grouped best practices to RadioGroup/CheckboxGroup and added singular best practices to Radio/Checkbox.

File: src/website/app/demos/Checkbox/index.js [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+    propsComment: (
+      <p>
+        Unlike most other components, which apply undocumented properties to the
+        root element, Checkbox applies undocumented properties to the{' '}
+        <em>input</em>.
+      </p>
+    ),
+    slug: 'checkbox',
+    title: 'Checkbox',
+    whenHowToUse: `Use a Checkbox to allow users to select one or more options
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/index.js]: Updated.

File: src/website/app/demos/Checkbox/index.js [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+from a list.
+
+Use Checkboxes to change settings rather than initiate actions.
+
+Mineral UI also provides a [CheckboxGroup](../checkbox-group), which offers a simpler
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/Checkbox/index.js]: Removed.

File: src/website/app/demos/routes.js [hunk 20]
```diff
@@ -74,6 +74,18 @@ export default {
     slug: 'card-title',
     title: 'CardTitle'
   },
+  checkbox: {
+    description:
+      'Checkboxes allows users to select one or more options from a list. Use Checkboxes to accept multiple choice input from a user.',
```
[the remark this answers is not in the document]
Comment (brentertz) [reply src/website/app/demos/routes.js]: Updated.

File: src/Checkbox/CheckboxGroup.js [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+    )
+  };
+});
+
+/**
+ * CheckboxGroup allows authors to construct a group of
+ * [Checkboxes](../checkbox) and provides a simpler API than working with
+ * Checkbox directly.
+ *
+ * Checkbox allows users to select multiple options from a list.
```
Comment (kylegach) [inline src/Checkbox/CheckboxGroup.js]: "_CheckboxGroup_ allows users..." ?

File: src/Checkbox/Checkbox.js [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+        name: 'Choice',
+        theme: {}
+      },
+      baseTheme
+    )
+  };
+});
+
+/**
+* Checkbox is an interactive control that can be turned on or off. Checkboxes
```
Comment (kylegach) [inline src/Checkbox/Checkbox.js]: "...can be _toggled_ on or off...? (Also applies to Radio)

Then again, if we'll eventually have a Toggle component, maybe it's best as-is. 🤷‍♂️

File: src/Radio/RadioGroup.js [hunk 23]
```diff
@@ -15,131 +15,90 @@
...
+  };
 });
 
 /**
- * RadioGroup allows users to select a single option from a set and provides a
- * simpler API than working with [Radios](../radio) directly.
+ * RadioGroup allows authors to construct a group of [Radios](../radio)
+ * and provides a simpler API than working with Radio directly.
+ *
+ * RadioGroup allows users to select a single option from a set.
```
Comment (kylegach) [inline src/Radio/RadioGroup.js]: We used "list" rather than "set" in CheckboxGroup.

Comment (brentertz) [reply src/Radio/RadioGroup.js]: Updated.

File: src/Checkbox/CheckboxGroup.js (hunk 21 above)
Comment (brentertz) [reply src/Checkbox/CheckboxGroup.js]: Updated.

File: src/Checkbox/Checkbox.js (hunk 22 above)
Comment (brentertz) [reply src/Checkbox/Checkbox.js]: 🤷‍♂️

**Pull request merged**